beta.go

```diff
@@ -0,0 +1,58 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+)
+
+// Beta returns the value of the complete beta function B(a, b).
```

Comment:
I started with `scipy.special.beta` to get my edge cases (which is using [cephes-beta](https://github.com/scipy/scipy/blob/v0.18.0/scipy/special/cephes/beta.c))

see:

```python
>>> import scipy
>>> scipy.version.version
'0.18.0'
>>> import scipy.special as spe
>>> tests = [(0,0),(1,0),(-1,0),(0,1),(0,-1)]
>>> for a,b in tests: print "beta(%+d, %+d) => %f" % (a,b,spe.beta(a,b))
... 
beta(+0, +0) => inf
beta(+1, +0) => inf
beta(-1, +0) => inf
beta(+0, +1) => inf
beta(+0, -1) => -inf
```

but then switched to `octave`.

indeed, `octave` has different edge cases:
```octave
$> octave-cli
GNU Octave, version 4.0.3
octave:1> beta(0,0)
ans = NaN
octave:2> beta(1,0)
ans = Inf
octave:3> beta(-1,0)
ans = NaN
octave:4> beta(0,1)
ans = Inf
octave:5> beta(0,-1)
ans = NaN
```

For continuity reasons, I suppose the `scipy` behaviour makes actually more sense (except for the `v < 0` cases where a `NaN` makes more sense):

```python
>>> atests = [(1e-18,1e-18),(1,1e-18),(-1,1e-18),(1e-18,1),(1e-18,-1)]
>>> for a,b in atests: print "beta(%+d, %+d) => %g" % (a,b,spe.beta(a,b))
beta(+0, +0) => 2e+18
beta(+1, +0) => 1e+18
beta(-1, +0) => inf
beta(+0, +1) => 1e+18
beta(+0, -1) => inf
```

```octave
octave:17> beta(1e-18,1e-18)
ans =    2.0000e+18
octave:18> beta(1,1e-18)
ans =    1.0000e+18
octave:19> beta(-1,1e-18)
ans = NaN
octave:20> beta(1e-18,1)
ans =    1.0000e+18
octave:21> beta(1e-18,-1)
ans = NaN
```

although, as we define `Beta` to be `Γ(a)Γ(b) / Γ(a+b)` and as `math.Gamma` has this:

```
func Gamma(x float64) float64
    Gamma returns the Gamma function of x.

    Special cases are:

	Gamma(+Inf) = +Inf
	Gamma(+0) = +Inf
	Gamma(-0) = -Inf
	Gamma(x) = NaN for integer x < 0
	Gamma(-Inf) = NaN
	Gamma(NaN) = NaN
```

returning `NaN` for `(0,0)` makes also sense...

I have also looked at `boost::math`:  http://www.boost.org/doc/libs/1_61_0/libs/math/doc/html/math_toolkit/sf_beta/beta_function.html and its `C++17` implementation: http://en.cppreference.com/w/cpp/numeric/special_math/beta

and also at GSL: https://www.gnu.org/software/gsl//manual/html_node/Beta-Functions.html#Beta-Functions

(the `C/C++` libraries just raise an error/abort when `v<=0`)

so... to conclude: OK to document the `(0,0)` special case to return `NaN` ?